stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -494,11 +494,9 @@ rdiv!(A::AbstractMatrix{T}, transD::Transpose{<:Any,<:Diagonal{T}}) where {T} =
 end
 
 function kron(A::Diagonal{T1}, B::Diagonal{T2}) where {T1<:Number, T2<:Number}
-    valA = A.diag; nA = length(valA)
-    valB = B.diag; nB = length(valB)
-    valC = Vector{typeof(zero(T1)*zero(T2))}(undef,nA*nB)
-    C = Diagonal(valC)
-    return @inbounds kron!(C, A, B)
+    Cdiag = kron(A.diag, B.diag)
+    C = Diagonal(Cdiag)
+    return C
 end
```

Comment:
Could be simplified to

```suggestion
kron(A::Diagonal{<:Number}, B::Diagonal{<:Number}) = Diagonal(kron(A.diag, B.diag))
```

Reply:
Updated to use this form